### PR TITLE
ida: Add IDA 7.2+ compatibility

### PR DESCRIPTION
**Pull request by Grazfather**
### Description/Motivation/Screenshots ###
This fixes up ida_gef.py to work with IDA > 7.2, while maintaining backwards compatibility.

### How Has This Been Tested? ###

| Architecture |          Yes/No          | Comments                                  |
| ------------ | :----------------------: | ----------------------------------------- |
| x86-32       | :heavy_check_mark: | Replace with :heavy_check_mark: if tested |
| x86-64       | :heavy_check_mark: |                                           |
| ARM          | :heavy_multiplication_x: |                                           |
| AARCH64      | :heavy_multiplication_x: |                                           |
| MIPS         | :heavy_multiplication_x: |                                           |
| POWERPC      | :heavy_multiplication_x: |                                           |
| SPARC        | :heavy_multiplication_x: |                                           |
| RISC-V       | :heavy_multiplication_x: |                                           |
| `make tests` | :heavy_multiplication_x: |                                           |

### Checklist ###

- [x] My PR was done against the `dev` branch, not `master`.
- [x] My code follows the code style of this project.
- [] My change includes a change to the documentation, if required.
- [] My change adds tests as appropriate.
- [x] I have read and agree to the **CONTRIBUTING** document.

- [x] improve docs

### CODE REVIEW

**Comment by Grazfather** [comment]
Big thanks to @jbcayrou for getting the ball rolling and pointing to an implementation that fixes the thread safety issues (In #510). I added an API mapping so that we can maintain compatibility with other IDA.

File: scripts/ida_gef.py [hunk 1]
```diff
@@ -131,106 +209,106 @@ def shutdown(self):
 
     @expose
     def makecomm(self, address, comment):
-        """ MakeComm(int addr, string comment) => None
+        """ makecomm(int addr, string comment) => None
         Add a comment to the current IDB at the location `address`.
         Example: ida MakeComm 0x40000 "Important call here!"
```
Comment (hugsy) [inline scripts/ida_gef.py:214]: Example needs to be updated

```
ida makecomm 0x40000 "Important call here!"
```

File: scripts/ida_gef.py [hunk 2]
```diff
@@ -131,106 +209,106 @@ def shutdown(self):
...
         addr = long(address, 16) if ishex(address) else long(address)
-        return idc.MakeComm(addr, comment)
+        return api.MakeComm(addr, comment)
 
     @expose
     def setcolor(self, address, color="0x005500"):
-        """ SetColor(int addr [, int color]) => None
+        """ setcolor(int addr [, int color]) => None
         Set the location pointed by `address` in the IDB colored with `color`.
         Example: ida SetColor 0x40000
```
Comment (hugsy) [inline scripts/ida_gef.py:223]: Same here (example needs to be updated)

File: scripts/ida_gef.py [hunk 3]
```diff
@@ -131,106 +209,106 @@ def shutdown(self):
...
         color = long(color, 16) if ishex(color) else long(color)
-        return idc.SetColor(addr, CIC_ITEM, color)
+        return api.SetColor(addr, CIC_ITEM, color)
 
     @expose
     def makename(self, address, name):
-        """ MakeName(int addr, string name]) => None
+        """ makename(int addr, string name]) => None
         Set the location pointed by `address` with the name specified as argument.
         Example: ida MakeName 0x4049de __entry_point
```
Comment (hugsy) [inline scripts/ida_gef.py:233]: Same here

File: scripts/ida_gef.py [hunk 4]
```diff
@@ -131,106 +209,106 @@ def shutdown(self):
...
         return None
 
     @expose
-    def importstruct(self, struct_name):
-        """ ImportStruct(string name) => dict
+    def importstruct(self, name):
+        """ importstruct(string name) => dict
         Import an IDA structure in GDB which can be used with the `pcustom`
         command.
         Example: ida ImportStruct struct_1
```
Comment (hugsy) [inline scripts/ida_gef.py:258]: Again

File: scripts/ida_gef.py [hunk 5]
```diff
@@ -131,106 +209,106 @@ def shutdown(self):
...
+            name: [x for x in api.StructMembers(struct)]
+        }
 
     @expose
     def importstructs(self):
-        """ ImportStructs() => dict
+        """ importstructs() => dict
         Import all structures from the current IDB into GDB, to be used with the `pcustom`
         command.
         Example: ida ImportStructs
```
Comment (hugsy) [inline scripts/ida_gef.py:272]: Yes here too

File: scripts/ida_gef.py [hunk 6]
```diff
@@ -131,106 +209,106 @@ def shutdown(self):
...
         addr = long(address, 16) if ishex(address) else long(address)
-        return idc.MakeName(addr, name)
+        return api.MakeName(addr, name)
 
     @expose
     def jump(self, address):
-        """ Jump(int addr) => None
+        """ jump(int addr) => None
         Move the IDA EA pointer to the address pointed by `addr`.
         Example: ida Jump 0x4049de
```
Comment (hugsy) [inline scripts/ida_gef.py:242]: Again

File: scripts/ida_gef.py (hunk 1 above)
Comment (Grazfather) [reply scripts/ida_gef.py:214]: The case doesn't matter.

Comment (hugsy) [reply scripts/ida_gef.py]: It is just a matter of consistency

**Pull request merged**